internal/protocoltest/awsrestjson/deserializers.go

```diff
@@ -886,70 +591,221 @@ func awsRestjson1_deserializeHttpBindingsHttpPayloadTraitsWithMediaTypeOutput(v
-	return "awsRestjson1_deserializeOpQueryIdempotencyTokenAutoFill"
-}
+		case "trueBooleanValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var b *bool
+			if val != nil {
+				jtv, ok := val.(bool)
+				if !ok {
+					return fmt.Errorf("expected TrueBooleanValue to be of type *bool, got %T instead", val)
+				}
+				b = &jtv
+			}
+			(*v).TrueBooleanValue = b
 
-func (m *awsRestjson1_deserializeOpQueryIdempotencyTokenAutoFill) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
-	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
-) {
-	out, metadata, err = next.HandleDeserialize(ctx, in)
-	if err != nil {
-		return out, metadata, err
-	}
+		case "falseBooleanValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var b *bool
+			if val != nil {
+				jtv, ok := val.(bool)
+				if !ok {
+					return fmt.Errorf("expected FalseBooleanValue to be of type *bool, got %T instead", val)
+				}
+				b = &jtv
+			}
+			(*v).FalseBooleanValue = b
 
-	response, ok := out.RawResponse.(*smithyhttp.Response)
-	if !ok {
-		return out, metadata, &smithy.DeserializationError{Err: fmt.Errorf("unknown transport type %T", out.RawResponse)}
-	}
+		case "byteValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var st *int8
+			if val != nil {
+				jtv, ok := val.(json.Number)
+				if !ok {
+					return fmt.Errorf("expected ByteValue to be json.Number, got %T instead", val)
+				}
+				i64, err := jtv.Int64()
+				if err != nil {
+					return err
+				}
+				st = ptr.Int8(int8(i64))
+			}
+			(*v).ByteValue = st
 
-	if response.StatusCode < 200 || response.StatusCode >= 300 {
-		errorType := response.Header.Get("X-Amzn-Errortype")
-		errorType = restjson.SanitizeErrorCode(errorType)
+		case "shortValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var st *int16
+			if val != nil {
+				jtv, ok := val.(json.Number)
+				if !ok {
+					return fmt.Errorf("expected ShortValue to be json.Number, got %T instead", val)
+				}
+				i64, err := jtv.Int64()
+				if err != nil {
+					return err
+				}
+				st = ptr.Int16(int16(i64))
+			}
+			(*v).ShortValue = st
 
-		decoder := json.NewDecoder(response.Body)
-		decoder.UseNumber()
-		defer response.Body.Close()
+		case "integerValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var st *int32
+			if val != nil {
+				jtv, ok := val.(json.Number)
+				if !ok {
+					return fmt.Errorf("expected IntegerValue to be json.Number, got %T instead", val)
+				}
+				i64, err := jtv.Int64()
+				if err != nil {
+					return err
+				}
+				st = ptr.Int32(int32(i64))
+			}
+			(*v).IntegerValue = st
+
+		case "longValue":
+			val, err := decoder.Token()
+			if err != nil {
+				return err
+			}
+			var st *int64
+			if val != nil {
+				jtv, ok := val.(json.Number)
```

Comment:
Also outside the scope of this PR, but  these `if val != nil` checks could be simplified a little if wanted so that they return early instead of intended.

```go
var st *int64
if val == nil {
    continue
}
// get value
(*v).LongValue = st
```

Reply:
Ah good call, as this is generated within the context of a loop

Reply:
Looking at this more, we would require a bit more changes to the deserializer if we wanted to simplify this further. As calling continue is not always valid unless there is further inspection on the context of the parent caller.